src/Stormpath/DataStore/DefaultDataStore.php

```diff
@@ -26,20 +28,24 @@
 use Stormpath\Resource\ResourceError;
 use Stormpath\Util\Version;
 
-class DefaultDataStore implements InternalDataStore
+class DefaultDataStore extends Cacheable implements InternalDataStore
 {
 
     private $requestExecutor;
     private $resourceFactory;
     private $baseUrl;
+    protected $cacheManager;
+    protected $cache;
 
     const DEFAULT_SERVER_HOST = 'api.stormpath.com';
     const DEFAULT_API_VERSION = '1';
 
-    public function __construct(RequestExecutor $requestExecutor, $baseUrl = null)
+    public function __construct(RequestExecutor $requestExecutor, CacheManager $cacheManager, $baseUrl = null)
     {
         $this->requestExecutor = $requestExecutor;
         $this->resourceFactory = new DefaultResourceFactory($this);
+        $this->cacheManager = $cacheManager;
+        $this->cache = $this->cacheManager->getCache();
```

Comment:
A Cache instance should not be created in the constructor. It should be retrieved/created on every request.